dev/service-registry/play-integration-javadsl/src/main/scala/com/lightbend/lagom/play/LagomPlayModule.scala

```diff
@@ -4,58 +4,85 @@
+    val triedAcls: Try[JList[_ <: Config]] = Try(config.underlying.getConfigList("lagom.play.acls"))
+
+    val warning = "Service setup via 'application.conf' is deprecated. Remove 'lagom.play.service-name' and/or " +
+      "'lagom.play.acls' and use 'bindServiceInfo' on your Play's Module class."
+
+    val maybeServiceInfoBinding = (triedServiceName, triedAcls) match {
+      case (Success(serviceName), Success(aclList)) => {
+        logger.warn(warning)
+        // create a ServiceInfo in case user doesn't see the warning
+        val acls = parseAclList(aclList)
+        Some(bind[ServiceInfo].toInstance(ServiceInfo.of(serviceName, acls: _*)))
+      }
+      case (Success(serviceName), _) => {
+        logger.warn(warning)
+        // create a ServiceInfo in case user doesn't see the warning
+        Some(bind[ServiceInfo].toInstance(ServiceInfo.of(serviceName)))
+      }
+      case (_, Success(_)) => {
+        logger.warn(warning)
+        // can't create a ServiceInfo because service-name is missing
+        None
+      }
+      case _ => None
+    }
+    maybeServiceInfoBinding
+  }
+
+  private def parseAclList(aclList: JList[_ <: Config]): Seq[ServiceAcl] = {
+    aclList.asScala.map { aclConfig =>
+      val method = if (aclConfig.hasPath("method")) {
+        Optional.of(new Method(aclConfig.getString("method")))
+      } else Optional.empty[Method]
+      val pathRegex = if (aclConfig.hasPath("path-regex")) {
+        Optional.of(aclConfig.getString("path-regex"))
+      } else Optional.empty[String]
+      new ServiceAcl(method, pathRegex)
+    }
   }
 }
 
 class PlayRegisterWithServiceRegistry @Inject() (config: Configuration, serviceInfo: ServiceInfo, serviceRegistry: ServiceRegistry, applicationLifecycle: ApplicationLifecycle) {
-
   private val httpAddress = config.underlying.getString("play.server.http.address")
   private val httpPort = config.underlying.getString("play.server.http.port")
   private val serviceUrl = new URI(s"http://$httpAddress:$httpPort")
 
-  private val acls = config.underlying.getConfigList("lagom.play.acls").asScala.map { aclConfig =>
-    val method = if (aclConfig.hasPath("method")) {
-      Optional.of(new Method(aclConfig.getString("method")))
-    } else Optional.empty[Method]
-    val pathRegex = if (aclConfig.hasPath("path-regex")) {
-      Optional.of(aclConfig.getString("path-regex"))
-    } else Optional.empty[String]
-
-    new ServiceAcl(method, pathRegex)
-  }
-  private val service = new ServiceRegistryService(serviceUrl, acls.asJava)
+  // TODO: ServiceRegistryService should not flatmap the ACL lists (locatableService's names are lost)
+  private val serviceAcls = serviceInfo.getLocatableServices.values().asScala.flatMap(_.asScala).toSeq.asJava
+  private val service = new ServiceRegistryService(serviceUrl, serviceAcls)
+  // TODO: fix -> this register operation is registering all ACLs under the microservice name, not under each locatable service name. Will lead to unlocatable.
```

Comment:
This TODO will be addressed in a separate PR. 

Play's ACLs don't need to be `locatable` in the sense that a Play's `ServiceInfo`  is not published on a ServiceRegistry, but Play's ACLs need to be `locatable` since that is also controlling if an ACL is published on the Service Gateway.

> Maybe we should more explicitly separate the concepts of proxying vs. location in the development gateway?

👍